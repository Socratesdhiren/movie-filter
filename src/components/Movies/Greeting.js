import { useState } from "react";

const Greeting = () => {
  const [isChange, setIsChange] = useState(false);

  const changeHandler = () => {
    setIsChange(true);
  };
  return (
    <div style={{ display: "flex", justifyContent: "center", margin: " 30px" }}>
      <h2>Hello Unit Test World!</h2>
      {!isChange && <p>It's glad to see you!</p>}
      {isChange && <p>Changed!</p>}

      <button onClick={changeHandler}>Change text!</button>
    </div>
  );
};

export default Greeting;
