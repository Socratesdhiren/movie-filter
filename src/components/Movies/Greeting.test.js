import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from 'react-test-renderer';
import Greeting from "./Greeting";

describe('Greeting componets', ()=>{
    test("Render Hello World", () => {
        // arrange
      
        render(<Greeting />);
      
        //Act

        // nothing

       
        //assert
      
        const helloWorldElement = screen.getByText("Hello Unit Test World!");
      
        expect(helloWorldElement).toBeInTheDocument();
      });

      test('renders good to see you if the button was not click', ()=>{
        render(<Greeting/>)
        const outputElement = screen.getByText("glad to see you", {exact: false} );
        expect(outputElement).toBeInTheDocument();


      });

      test('render change if the button was click', () =>{
        //arrange
        render(<Greeting/>)
        // Act

        const buttonElement = screen.getByRole('button')
        //        const buttonElement = screen.getByText('button')

        userEvent.click(buttonElement)
      
        // Assert
        const outputElement = screen.getByText("Changed!");
        expect(outputElement).toBeInTheDocument();
      });

      test('doesnot render "glad to see you if button was click" ', ()=>{
         //arrange
         render(<Greeting/>)
         // Act
 
         const buttonElement = screen.getByRole('button')
         //        const buttonElement = screen.getByText('button')
 
         userEvent.click(buttonElement)

         const outputElement = screen.queryByText("glad to see you", {exact: false} );
         //         const outputElement = screen.getByText("glad to see you", {exact: false} );

         expect(outputElement).toBeNull();


      })
})


describe('SnapShot matches', ()=>{
    it('renders correctly', () => {
        const tree = renderer
          .create(<Greeting/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
})



