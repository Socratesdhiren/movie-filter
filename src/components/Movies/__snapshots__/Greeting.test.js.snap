// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot matches renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
      "margin": " 30px",
    }
  }
>
  <h2>
    Hello Unit Test World!
  </h2>
  <p>
    It's glad to see you!
  </p>
  <button
    onClick={[Function]}
  >
    Change text!
  </button>
</div>
`;
